pkg/storage/chunk/chunk_store_utils.go

```diff
@@ -162,7 +214,10 @@ func (c *Fetcher) FetchChunks(ctx context.Context, chunks []Chunk, keys []string
 	}
 
 	// Always cache any chunks we did get
-	if cacheErr := c.writeBackCache(ctx, fromStorage); cacheErr != nil {
+	if cacheErr := c.writeBackCacheAsync(fromStorage); cacheErr != nil {
+		if cacheErr == errMemcachedAsyncBufferFull {
```

Comment:
I'm not sure we need a metric here; would a log not suffice?

I'm not sure that `loki_chunk_fetcher_cache_skipped_total` accurately reflects what's happening here. We're skipping a write-back operation because a buffer is full, while the description `"Total number of operations against cache that have been skipped."` is a bit too general in this case. If we're going to keep this metric, I think we need to be more specific.